### PR TITLE
Moar promises

### PR DESCRIPTION
**Pull request by fippo**
Lets dogfood on that promises stuff...

I found a really odd bug... createAnswers gets a third constraints argument which doesn't make sense.
@alvestrand is that worth bailing out on in adapter?

### CODE REVIEW

File: src/web_app/js/peerconnectionclient.js [hunk 1]
```diff
@@ -174,9 +174,9 @@ PeerConnectionClient.prototype.getPeerConnectionStats = function(callback) {
 
 PeerConnectionClient.prototype.doAnswer_ = function() {
   trace('Sending answer to peer.');
-  this.pc_.createAnswer(this.setLocalSdpAndNotify_.bind(this),
-      this.onError_.bind(this, 'createAnswer'),
-      PeerConnectionClient.DEFAULT_SDP_CONSTRAINTS_);
+  this.pc_.createAnswer()
```
Comment (jiayliu) [inline src/web_app/js/peerconnectionclient.js]: The answer constraints should not be removed, webrtc looks for it to decide if it receives media or uses bundle:

https://code.google.com/p/chromium/codesearch#chromium/src/third_party/libjingle/source/talk/app/webrtc/mediastreamsignaling.cc&rcl=1437050479&l=442

Comment (fippo) [inline src/web_app/js/peerconnectionclient.js]: hrm... that should hopefully work with https://github.com/webrtc/adapter/blob/master/adapter.js#L228-229
now if firefox doesn't error i can fix that.

Comment (fippo) [inline src/web_app/js/peerconnectionclient.js]: Firefox throws an error unfortunately :-(

**Comment by fippo** [comment]
@jiayliu worked around the issue -- ugly but *shrug*

I've also seen getStats failures since apprtc is using the legacy syntax without a browser switch but that's a different story

**Comment by jiayliu** [comment]
Does it mean we have a regression in infobox on firefox?

**Comment by fippo** [comment]
I doubt it's a regression. Firefox getStats has never worked without a track selector as its called in https://github.com/webrtc/apprtc/blob/master/src/web_app/js/peerconnectionclient.js#L172

**Comment by jiayliu** [comment]
lgtm.
ok to merge?

**Comment by fippo** [comment]
yeah. The Firefox issue should soon be fixed but I guess it will take time to go to the stable channel there.

**Pull request merged**